### ISSUE TITLE
IllegalStateException thrown after resetting player and setting data source

### ISSUE THREAD

**Issue by velocitysystems** [protyposis/ITEC-MediaPlayer#114]
### Description

MediaPlayerExtended throws an IllegalStateException when switching between data sources which _contain_ and _do not contain_ an audio track.

### Steps to Reproduce
1. Set data source to media containing a video track **with an audio track.**
2. Call the `prepare()` or `prepareAsync()` method.
3. Call the `reset()` method.
4. Set data source to media containing a video track **without an audio track.**
5. Call the `prepare()` or `prepareAsync()` method.

### Expected Behavior
The media should be prepared for playback.

### Actual Behavior
An `IllegalStateException` is thrown.

```
java.lang.IllegalStateException
02-13 11:03:01.878 E/MediaPlayer(12568): 	at net.protyposis.android.mediaplayer.AudioPlayback.pause(AudioPlayback.java:205)
02-13 11:03:01.878 E/MediaPlayer(12568): 	at net.protyposis.android.mediaplayer.MediaPlayer.prepareInternal(MediaPlayer.java:456)
02-13 11:03:01.878 E/MediaPlayer(12568): 	at net.protyposis.android.mediaplayer.MediaPlayer.access$800(MediaPlayer.java:36)
02-13 11:03:01.878 E/MediaPlayer(12568): 	at net.protyposis.android.mediaplayer.MediaPlayer$PlaybackThread.prepareInternal(MediaPlayer.java:1101)
02-13 11:03:01.878 E/MediaPlayer(12568): 	at net.protyposis.android.mediaplayer.MediaPlayer$PlaybackThread.handleMessage(MediaPlayer.java:1053)
02-13 11:03:01.878 E/MediaPlayer(12568): 	at android.os.Handler.dispatchMessage(Handler.java:98)
02-13 11:03:01.878 E/MediaPlayer(12568): 	at android.os.Looper.loop(Looper.java:154)
02-13 11:03:01.878 E/MediaPlayer(12568): 	at android.os.HandlerThread.run(HandlerThread.java:61)
```

### Discussion

I believe the issue is that `mAudioPlayback` is never set to `null` when calling `reset()` . Since the first data source contained an audio track, this field was set but never nulled out when calling `reset()`or setting the second data source which does not contain an audio track.

https://github.com/protyposis/MediaPlayer-Extended/blob/2ff8e6db7ca7861efe7c81bc577b856efc27999e/MediaPlayer/src/main/java/net/protyposis/android/mediaplayer/MediaPlayer.java#L456

**Comment by apwillies** [comment]
@protyposis Any chance of getting this one fixed soon?

**Comment by protyposis** [comment]
Thanks for reporting this issue. I'll try to take a look in the next few days.

**Comment by protyposis** [comment]
Fixed in v4.4.1.
